### ISSUE TITLE
FR: 

### ISSUE THREAD

**Issue by HSUCHING** [zsviczian/obsidian-excalidraw-plugin#1674]
Feature request:
     In excalidraw, I may need a function similar to templates and deep copying. For example, in my entire canvas, for some elements that need to be used multiple times, these elements may be composed of SVG, PNG, or combinations of images and text. I need to use them multiple times on the canvas to express the same relationships and so on. Right now I can only do shallow copies for each one, but if there is any change, I have to make changes everywhere once again which is very troublesome. Is there a better solution?

Desired Solution:
I hope to have a feature that helps me create something like a template, and then deeply duplicate it to every place I need to use it. If one day I change the original template, all the references can be updated directly to the latest style. Of course, if there are some parameters that can be set in the original template, allowing some flexibility for changes in the referenced places would be best. Of course, I know this request is a bit high.

PS:
Perhaps this plugin already has similar functionality. If it does, please let me know how to use them.

**Comment by zsviczian** [comment]
1) I think what you might be looking for is deconstruction. Take a look at this: https://youtu.be/HRtaaD34Zzg , and this: https://youtu.be/mvMQcz401yo
My logic with deconstruction is to reuse parts of images in various illustrations. I have some components that are by now 5-6 layers deep... because as I progress with notes I end up deconstructing elements out of deconstructed parts to reuse and to connect with other notes. 

2) Else, maybe framing the relevant items you want to deep copy could be a solution.  You can activate the frame tool with the "F" shortcut

**Comment by HSUCHING** [comment]
@zsviczian Yes. I get it.This method "deconstruction" can really help me solve some problems. But I think there is a limitation in this aspect, that is, the premise is that the reuse part should be large enough, or a complete theme.It may be more useful. But I actually prefer a smaller granularity reuse. For example, in order to clearly express the "relationship" between the two concepts, I will express it with some icons/svg: A is a prerequisite for B. B is the basic principle of C. Or, list a style to indicate the inclusion relationship between E and F. At such a granularity level, it seems that this method can't help me, because if I want to do it, it is equivalent to deconstructing every "relationship" to form a drawing board, which is a huge work. My usage scenario is: When I build my notes, I actually want to formalize the "relationship" between each entity in a more fixed way, and because in the initial construction, the visual relationship that may be used will often change because I see some more appropriate ways, so I hope to visualize it. After the style is changed, the overall layout is refreshed.
